code/src/cljs/sixsq/nuvla/ui/edges/events.cljs

```diff
@@ -532,3 +546,56 @@
     {:storage/set {:session? false
                    :name     spec/local-storage-key
                    :value    (merge (edn/read-string storage) preference)}}))
+(reg-event-fx
+  ::set-selected-fleet-timespan
+  (fn [{db :db} [_ timespan]]
+    (let [{:keys [from to]} timespan]
+      {:db (assoc db ::spec/fleet-timespan timespan)
+       :fx [[:dispatch [::fetch-fleet-stats {:from        from
+                                             :to          to
+                                             :granularity (ts-utils/granularity-for-timespan timespan)
+                                             :dataset     fleet-availability-stats}]]]})))
+
+(reg-event-fx
+  ::fetch-fleet-stats
+  (fn [{{:keys [::spec/nuvlaboxes] :as db} :db} [_ {:keys [from to granularity dataset]}]]
+    (let [filter-str (if (seq (:resources nuvlaboxes))
+                       (->> nuvlaboxes
+                            (:resources)
+                            (mapv :id)
+                            (map (fn [ne-id]
```

Comment:
instead of building a huge filter with id= or id=...
you can use id=[id1 id2]. There is an utility for that here:
sixsq.nuvla.ui.utils.general/filter-eq-vals